src/Simulator/IncrementalRedstoneSimulator/ForEachSourceCallback.cpp

```diff
@@ -78,6 +59,53 @@ void ForEachSourceCallback::operator()(Vector3i Location)
 
 
 
+void ForEachSourceCallback::CheckIndirectPower()
+{
+	const Vector3i OffsetYP(0, 1, 0);
+	const auto Above = m_Position + OffsetYP;
+
+	if (Above.y == cChunkDef::Height)
+	{
+		return;
+	}
+
+	// Object representing restarted power calculation where the
+	// block above this piston, dropspenser is requesting a power level.
+	ForEachSourceCallback QuasiQueryCallback(m_Chunk, Above, m_Chunk.GetBlock(Above));
+
+	// Manually feed the callback object all positions that may deliver power to Above:
+	for (const auto QuasiPowerOffset : cSimulator::GetLinkedOffsets(OffsetYP))
+	{
+		QuasiQueryCallback(m_Position + QuasiPowerOffset);
+	}
+
+	// Get the results:
+	Power = std::max(Power, QuasiQueryCallback.Power);
+}
+
+
+
+
+
+bool ForEachSourceCallback::ShouldQueryLinkedPosition(const BLOCKTYPE Block)
+{
+	switch (Block)
+	{
+		// Normally we don't ask solid blocks for power because they don't have any (store, dirt, etc.)
+		// However, these are mechanisms that are IsSolid, but still give power. Don't ignore them:
+		case E_BLOCK_BLOCK_OF_REDSTONE:
```

Comment:
There is a problem with those, as the block they power via QC needs an update to actually be powered.
[As shown here](https://minecraft.gamepedia.com/Tutorials/Quasi-connectivity#Update_QC_activation)
And the same goes for attachable redstone components, unless they are attached to the block above the piston, because:
>The following redstone components can activate mechanism components one block away, and update redstone components adjacent to the block they are attached to (including above and below) as well as redstone components adjacent to themselves: 
 >  * Buttons (attaches in any direction)
 > * Detector Rail (attaches only downwards)
 >  * Lever (attaches in any direction)
 >  * Pressure Plates (attaches only downwards)
 >  * Trapped Chest (doesn't actually attach, but updates as if attached to block beneath)
 >  * Tripwire Hook (attaches only sideways)
 >  * Weighted Pressure Plates (attaches only downwards)

Which means they technically can provide power right away if there exists a directly powered redstone component that is adjacent to both the QC-powered component and the source, thus 'transferring' block update to the QC powered component, like this:
1. Without block transferring an update:
![2020-09-20_14 38 21](https://user-images.githubusercontent.com/60849082/93709424-86aff500-fb2d-11ea-8f30-3c80b9aa7159.png)
![2020-09-20_14 38 22](https://user-images.githubusercontent.com/60849082/93709421-857ec800-fb2d-11ea-9e2e-e161c111c263.png)

2. With an update-transferring block in between the source and the QC site:
![2020-09-20_14 38 31](https://user-images.githubusercontent.com/60849082/93709420-844d9b00-fb2d-11ea-954f-caf9cd1a577d.png)
![2020-09-20_14 38 34](https://user-images.githubusercontent.com/60849082/93709418-8283d780-fb2d-11ea-9a5c-65957361a7fe.png)

This is just what I've run into while tinkering with pistons and redstone piston handler. Hope you'll find anything of this useful.

Reply:
Thanks, if I understand you correctly, you're saying our block updates are too generous right? Yeah that's the case, currently most block changes update 24 (I think) redstone positions around it. That would need a separate fix.